crossbar/router/broker.py

```diff
@@ -171,13 +171,17 @@ def _filter_publish_receivers(self, receivers, publish):
 
         # if "eligible_authid" we only accept receivers that have the correct authid
         if publish.eligible_authid:
+            eligible = set()
             for aid in publish.eligible_authid:
-                receivers = receivers & self._router._authid_to_sessions.get(aid, set())
```

Comment:
I'm afraid @markope is right here. eligible needs to intersect the receivers with the unionset of all sets of session IDs mapped from authids/authroles.

Logically, this can be done in a loop as above. This definitely produces temporary sets over and over (within the loop body). Whether the following is producing less GC pressure (and hence is faster), I'm not sure, but here is a functional alternative:

```python
set.union(*[set([1,2,3]), set([2,3,4]), set([3,4,5])])
```